### ISSUE TITLE
Should Detector IOU Threshold be same as IOU threshold for mAP?  

### ISSUE THREAD

**Issue by takehiro-code** [Cartucho/mAP#140]
Hi, I have a question regarding the detector iou threshold and mAP iou threshold and I think it will be a simple question or just a clarification. I am using yolov3 detector with various confidence threshold and iou threshold and trying to get mAP for each detector configuration. My goal here is I am doing grid search of confidence threshold and iou threshold values based on maximizing mAP. My question here is should I set iou threshold value (default as `MINOVERLAP = 0.5`) same as the detector threshold? 

For example, I run detector with iou threshold 0.1, 0.3, 0.5, 0.7, then should the iou threshold be same accordingly as 0.1, 0.3, 0.5, 0.7 in the variable `MINOVERLAP` or does it have to be 0.5 to evaluate the detector performance for various iou threshold.

My assumption here is I should stick with 0.5 since I am measuring the detector performance for various iou threshold but decreasing `MINOVERLAP` means mAP will be higher, which negate the meaning of assesing the performance on various detector iou threshold.

**Comment by janedoesrepo** [comment]
You’re right. Detector iou is Independent of minoverlap. According to the pascal voc papers it is your responsibilty to filter the boxes before the evaluation.

**Comment by takehiro-code** [comment]
Thank you!

**Comment by takehiro-code** [comment]
@janedoesrepo Just to clarify, I seem not able to find the source that says it is our responsibility to filter the boxes before evaluation. I looked at the PASCAL VOC paper https://homepages.inf.ed.ac.uk/ckiw/postscript/ijcv_voc09.pdf but not able to find it, could you point out the source by any chance?

As I start to understand the mAP definition, I see people are computing mAP, setting the confidence threshold 0. This is because mAP is computed using the data points from the lowest confidence values to highest confidence values by setting the confidence threshold to 0 (which means technically detected target of confidence value 0 will be included in the mAP evaluation as well as confidence value of 1.0). Therefore, I wonder if it is logical to set a confidence threshold for reporting numbers, in my case 0.25 instead of 0, and mAP is computed using the data points above 0.25 to the highest confidence value at the detected target. Note that 0.25 is set for other purposes (for object tracking).

**Comment by janedoesrepo** [comment]
Hi @takehiro-code , you're right. It's in the Documentation of the VOC2007 (probably also in 2012) [Devkit](http://host.robots.ox.ac.uk/pascal/VOC/voc2007/devkit_doc_07-Jun-2007.pdf). There, in the evaluation section of the detection task, it says that "it is the responsibility of the participant’s system to filter multiple detections from its output." This usually the NMS part where you set your own IoU threshold.

When I evaluated Tiny YOLOv2 on VOC2007 I achieved the best result ~55% mAP with a confidence threshold of 0.005. However, this is utterly useless in practice because it results in many many False Positives. I guess this works because Recall does not consider False Positives. I wonder, however, if mAP is a good measure when the best results are achieved with totally impractical detections. When actually running my system I use a confidence threshold of 0.25. When evaluated this yielded around 33% mAP.
